Move SADL from Java 8 to Java 11

tuxji opened the pull request:
Tidy the Maven poms, change the Maven compiler release and bundle required execution environment from Java 8 to Java 11, and fix compilation errors to ensure SADL can be installed into Eclipse 2020-12 successfully.

Convert all files with DOS line endings to Unix line endings.

Remove some obsolete or unneeded files:
- sadl3/README.md
- all META-INF/services files
- .cvsignore
- javadoc.xml
- sadl3/deployment

Make sure all MANIFEST.MF files have:
- Bundle-RequiredExecutionEnvironment: JavaSE-11
- Bundle-Vendor: GE Research
- javax.activation replaced by jakarta.activation

Make sure all build.properties have their output lines removed since Tycho ignores it anyway.

Make sure all poms have:
- Same parent (sadl3/com.ge.research.sadl.parent/pom.xml)
- All versions moved to parent's dependencyManagement/pluginManagement
- All dependencies/plugins updated to latest versions
- All poms sorted and indented appropriately by sortpom plugin
- All plugin configurations and invocations simplified
- All unneeded parts removed

Make sure all Java files have:
- javax.activation replaced by jakarta.activation
- All compilation errors fixed
- Some deprecated calls fixed
- Some dead code removed
- Some warnings fixed

com.ge.research.sadl.target.target:
- Make all update sites match DSL 2020-12's update sites

tuxji committed:
Convert DOS line endings to Unix

tuxji committed:
Move SADL from Java 8 to Java 11

Tidy the Maven poms, change the Maven compiler release and bundle
required execution environment from Java 8 to Java 11, and fix
compilation errors to ensure SADL can be installed into Eclipse
2020-12 successfully.

Removed some obsolete or unneeded files:
- sadl3/README.md
- all META-INF/services files
- .cvsignore
- javadoc.xml
- sadl3/deployment

Made sure all MANIFEST.MF files had:
- Bundle-RequiredExecutionEnvironment: JavaSE-11
- Bundle-Vendor: GE Research
- javax.activation replaced by jakarta.activation

Made sure all build.properties had their output lines removed since
Tycho ignores it anyway.

Made sure all poms had:
- Same parent (sadl3/com.ge.research.sadl.parent/pom.xml)
- All versions moved to parent's dependencyManagement/pluginManagement
- All dependencies/plugins updated to latest versions
- All poms sorted and indented appropriately by sortpom plugin
- All plugin configurations and invocations simplified
- All unneeded parts removed

Made sure all Java files had:
- javax.activation replaced by jakarta.activation
- All compilation errors fixed
- Some deprecated calls fixed
- Some dead code removed
- Some warnings fixed

com.ge.research.sadl.target.target:
- all update sites updated to match 2020-12's update sites

crapo reviewed (changes requested):
@tuxji , I can successfully run a Maven install on the ji/update-poms branch but I have two errors in Eclipse. 

1. In the com.ge.research.jena project, the MANIFEST.MF file line 143 has export package org.pojava.datetime, giving error "Package 'org.pojava.datetime' does not exist in this plug-in".
2. The ...sadl.tests project, class TestJenaTranslator, references pojava class org.pojava.datetime.Duration but it cannot be resolved. Not sure of the right way to resolve this. 

The actual classes are in the datetime.jar file and somehow references to the classes are resolved except in the case of #2. Not sure how that happens as the datetime.jar file isn't a referenced library in #1.

General observations/comments:
1. Fuzzy on removal of the Jena n3 stuff. We still use n3, turtle but I guess not these packages anymore.
2. ...sadl.importer.ui.tests and ...sadl.testwuite.ui.tests should eventually have some tests. Work in progress on importer.
3. No need for copyright header in pom files?

Wow John!! This is a lot of cleanup! Thank you!

tuxji committed:
Fix Eclipse workspace problems

Ensure we can 1) run a Maven build on the SADL modules; 2) import the
SADL modules into an Eclipse 2021-03 workspace; 3) reload the target
platform; and 4) end up with no errors in the Eclipse workspace.

In com.ge.research.jena, remove pojava since SADL uses only datetime,
not both datetime and pojava.

In OwlToSadl.java, fix warning about questionable equals comparison
between String and Reasoner object, but there are some other warnings
in the same file that should be investigated too.

In poms, ensure used dependencies are declared and remove unused
dependencies.

In com.ge.research.sadl.tests, make sure MANIFEST.MF imports datetime.

In OntologyGraphGeneratorHandler.java and
ReasonerConfigurationDialog.java, remove unneeded @SuppressWarnings
lines.

Rename sadl_implicit_mode_fragment_provider.exsd to more correct name
sadl_implicit_model_fragment_provider.exsd ("model" was missing "l").

tuxji commented:
@crapo You're welcome!  I've made some changes to ensure  we can 1) run a Maven build on the SADL modules; 2) import the SADL modules into an Eclipse 2021-03 workspace; 3) reload the target platform; and 4) end up with no errors in the Eclipse workspace.  I also fixed a handful of warnings here and there such as a questionable equals comparison between String and Reasoner in OwlToSadl.java, but there are some other warnings in the same file that should be investigated and fixed too.

crapo commented:
@tuxji , Eclipse projects now show no errors after reloading target and updating Maven projects (provided that I do a final project clean/build on the three reasoner projects, which makes a pom error go away, but that was the case before as well) and the Maven build is successful as well. Looks good to me! Thanks.
